docs/contributing/Style-guide-for-Flutter-repo.md

```diff
@@ -1899,9 +1899,9 @@ The `dart:math` library is always imported `as math`.
 
 As a general rule, if a feature is entirely self-contained (not requiring low-level integration into the Flutter framework) and is not something with universal appeal, we would encourage that that feature be provided as a package.
 
-We try to be very conservative with what we put in the core framework, because there's a high cost to having anything there. We have to commit to supporting it for years to come, we have to document it, test it, create samples, we have to consider everyone's varied desires which they may have as they use the feature, we have to fix bugs. If there's design problems, we may not find out for a long time but then once we do we then have to figure out how to fix them without breaking people, or we have to migrate all our existing widgets to the new architecture, etc.
+We try to be conservative with what we add to the framework because we have to support it. We have to document it, test it, and create samples. We have to consider the many ways people may use it. We have to fix bugs. If there are design problems, we may not find out for a long time. Once we do, we have to figure out how to fix them without breaking downstream code, or we have to migrate all our existing widgets to the new design.
```

Comment:
This removes a lot of meaningful detail and changes nuances of meaning, which is unhelpful.

If a goal here was to shorten the paragraph, consider that the point of the paragraph is to say there's a lot we have to do over time after something is added to the core framework. If this terse summary of those things already feels like it's running on… well, that in itself helps make the point.

Reply:
I think this is a good point.
Personally, I'd like to retain the "If there's design problems" → "If there are design problems" grammar fix.

Reply:
I tried to tighten up this paragraph by removing unnecessary words and by demonstrating rather than telling the high support costs of framework code. Rereading both, I do prefer the revised over the original. What do you think? Is it a keeper?

Reply:
Personally, I'm a fan of how the original uses intentional run-on sentences to help get the point across.

But at the same time, I think parts of this wording change are a clear improvement over the original. When I hear "there's a high cost to having anything there", I assume that the author doesn't want anything there (and obviously we don't want to have an empty framework).

Perhaps the best of both worlds would be something like the following:

> We try to be very conservative with what we put in the core framework, because every feature we add is a lot of work to support. We have to document it, test it, create samples, we have to consider the varied desires that people may have as they use the feature, we have to fix bugs. If there are design problems, we may not find out for a long time, but then once we do we then have to figure out how to fix them without breaking downstream code, or we have to migrate all our existing widgets to the new architecture, etc.

Reply:
Yeah, in general one thing that's tricky about the "remove unnecessary words" vein of writing advice is that words that may seem unnecessary when using that lens often do have an effect on the sentence's meaning. For example here, "very", "core", "high cost", "years to come", "as they use the feature", "people" (vs "downstream code"), all carry part of the meaning, and the paragraph doesn't communicate all the same information when one of those is removed.

Some of them are redundant given the right background understanding of what it means to maintain a system… but for a reader who shares *all* the right background understanding, the whole paragraph is superfluous (as is lots of the rest of the document :wink:).

> When I hear "there's a high cost to having anything there", I assume that the author doesn't want anything there (and obviously we don't want to have an empty framework).

I see. I think the intended meaning here could be expressed more precisely as "there's a high cost to anything we put there". Does that prevent the unintended reading for you?

The high cost is real, but a high cost doesn't mean something isn't worth having — it just means there needs to be a corresponding benefit that outweighs the cost.

Reply:
In my mind, the phrase "breaking people" could mean any number of different things; I'd prefer swapping it out.

> there's a high cost to anything we put there

That sounds really good! Will edit the above comment :)